### DISCUSSION TITLE
Add new ARM type mortgage

### DISCUSSION THREAD

**Discussion by yellowbean** [yellowbean/AbsBox#10] [General]
with 
- periodic adj cap ( increase each reset time)
- life time cap  ( total increase cap )
- payment cap ( with negative amortization)

- init rate with init period
[201204_CFPB_ARMs-brochure.pdf](https://github.com/yellowbean/AbsBox/files/11329873/201204_CFPB_ARMs-brochure.pdf)

Probably new assumption at the end of initial period ends-> pct % of balance will repay if rate is higher .

ie.-> 
5/1, fix rate for 5 years and reset every 1 year afterwards
5/6, fix rate for 5 years and reset every 6 months afterwards

**Comment by yellowbean** [comment]
Complete with version  0.15.x

**Comment by kechenhk-lab** [comment]
Hi Shawn — following up on ARM support. I have been building a pure-Python port of Hastructure (pyabs) and need to cross-validate ARM cashflow projection against the reference engine. Currently `calcCashflow` for `AdjustRateMortgage` returns `Left "to be implement on adjust rate mortgage"` on master, Aug25, and Dec25 branches (Mortgage.hs:319).

Questions:
1. Is there an unpushed/private branch where ARM `calcCashflow` is implemented?
2. Is there a timeline for ARM support?
3. For projected-cashflow revolving pools (BMW test07 deal), the engine errors with `Can not divide zero on ds: OriginalPoolBalance Nothing` — is there a fix planned?

If reference output for either deal type becomes available, I can benchmark immediately. Thanks!

**Reply by yellowbean** [reply]
1. Probably not
2. If it is not supported, I'll fix this in the weekend
3. Sure, still target fixing it in this coming weekend, otherwise I'm losing the streak on Duolinguo

**Comment by yellowbean** [comment]
Hi, Moving the ETA of this end of week .

**Comment by yellowbean** [comment]
working on this, ETA in 3 days to be avaible in `Dec25` Branch

**Comment by yellowbean** [comment]
Test case [fix_arm.py](https://github.com/user-attachments/files/31350059/fix_arm.py) ( pls use `marimo edit fix_arm.py` to open it ) 

The issue resovled at 0.5.25 PROD engine
The sample case shall works with new syntax of schdule cashflow .
let me know if anything occured, thanks @kechenhk-lab
